### PR TITLE
Add `$schema` to infection.json5.dist, exclude files that breaks PHPUnit execution with Infection

### PR DESCRIPTION
**Pull request by maks-rafalko**
See https://github.com/sebastianbergmann/phpunit/pull/5788#issuecomment-2033186772 for the detailed explanation about what is going on and why it's needed.

### CODE REVIEW

**Comment by staabm** [comment]
Thank you 🙏

**Comment by sebastianbergmann** [comment]
I am thankful for your work, @maks-rafalko and @staabm, but I am still seeing issues such as this basically everywhere I look:

The [`Finished`](https://github.com/sebastianbergmann/phpunit/blob/b18d5f2fe3a1a0258f7b0de4fab2e32611dd87bc/src/Event/Events/Test/Lifecycle/Finished.php) class has a `public` method named `test`.

Infection's `PublicVisibility` mutator replaces `public` and `protected` and Infection then claims that this mutation survived, as can be seen in this screenshot:

318847058-26e1ad47-7f1b-4d26-9fda-17c344d12f0e

However, when I perform this mutation manually ...

```patch
diff --git a/src/Event/Events/Test/Lifecycle/Finished.php b/src/Event/Events/Test/Lifecycle/Finished.php
index 72032cf71..ae75d35c1 100644
--- a/src/Event/Events/Test/Lifecycle/Finished.php
+++ b/src/Event/Events/Test/Lifecycle/Finished.php
@@ -37,7 +37,7 @@ public function telemetryInfo(): Telemetry\Info
         return $this->telemetryInfo;
     }
 
-    public function test(): Code\Test
+    protected function test(): Code\Test
     {
         return $this->test;
     }
```

... and try to run PHPUnit's test suite then I get this error:

```
$ ./phpunit --testsuite unit
PHPUnit 11.1-gb18d5f2fe3 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.4
Configuration: /usr/local/src/phpunit/phpunit.xml

.

An error occurred inside PHPUnit.

Message:  Call to protected method PHPUnit\Event\Test\Finished::test() from scope PHPUnit\Runner\ResultCache\ResultCacheHandler
Location: /usr/local/src/phpunit/src/Runner/ResultCache/ResultCacheHandler.php:119

#0 /usr/local/src/phpunit/src/Runner/ResultCache/Subscriber/TestFinishedSubscriber.php(27): PHPUnit\Runner\ResultCache\ResultCacheHandler->testFinished()
#1 /usr/local/src/phpunit/src/Event/Dispatcher/DirectDispatcher.php(102): PHPUnit\Runner\ResultCache\TestFinishedSubscriber->notify()
#2 /usr/local/src/phpunit/src/Event/Dispatcher/DeferringDispatcher.php(45): PHPUnit\Event\DirectDispatcher->dispatch()
#3 /usr/local/src/phpunit/src/Event/Emitter/DispatchingEmitter.php(962): PHPUnit\Event\DeferringDispatcher->dispatch()
#4 /usr/local/src/phpunit/src/Framework/TestRunner.php(235): PHPUnit\Event\DispatchingEmitter->testFinished()
#5 /usr/local/src/phpunit/src/Framework/TestCase.php(337): PHPUnit\Framework\TestRunner->run()
#6 /usr/local/src/phpunit/src/Framework/TestSuite.php(369): PHPUnit\Framework\TestCase->run()
#7 /usr/local/src/phpunit/src/Framework/TestSuite.php(369): PHPUnit\Framework\TestSuite->run()
#8 /usr/local/src/phpunit/src/Framework/TestSuite.php(369): PHPUnit\Framework\TestSuite->run()
#9 /usr/local/src/phpunit/src/TextUI/TestRunner.php(62): PHPUnit\Framework\TestSuite->run()
#10 /usr/local/src/phpunit/src/TextUI/Application.php(200): PHPUnit\TextUI\TestRunner->run()
#11 /usr/local/src/phpunit/phpunit(104): PHPUnit\TextUI\Application->run()
#12 {main}
```

The mutation causes an error in PHPUnit's test runner that is not specific to a single test.

Sure, we can exclude mutators and/or source files. But I doubt that this will lead to a useful report. I fear that we must accept the fact that mutating the test (framework|runner) while running its test is an edge case that neither of us should worry about: Infection should not be adapted to cater to the mutation testing of PHPUnit, nor should PHPUnit be adapted so that mutation testing can be performed on it.

Maybe I am just too pessimistic here, but, as always, I love to be proven wrong.

In the near future (days/weeks, not months), I will add Infection to the dependencies of PHPUnit (see https://github.com/sebastianbergmann/phpunit/pull/5788#issuecomment-2031087799). This *should* be more straightforward.
